Sources/BSON/ByteBuffer+BSON.swift

```diff
@@ -1,30 +1,50 @@
 import NIO
 
 extension ByteBuffer {
-    /// Write null terminated string into this ByteBuffer using UTF-8 encoding,
-    /// moving the writer index forward by the byte length of string + 1 (for null terminator).
+    /// Write null terminated UTF-8 string to ByteBuffer starting at writerIndex
     @discardableResult
     internal mutating func writeCString(_ string: String) -> Int {
         let written = self.writeString(string + "\0")
         return written
     }
 
-    /// Read bytes off this ByteBuffer until encountering null, decoding it as String using the UTF-8 encoding.
-    /// moving the reader index forward by the byte length of string + 1 (for null terminator).
+    /// Attempts to read null terminated UTF-8 string from ByteBuffer starting at the readerIndex
     internal mutating func readCString() throws -> String {
-        var bytes: [UInt8] = []
-        for _ in 0..<BSON_MAX_SIZE {
-            guard let b = self.readBytes(length: 1) else {
-                throw BSONError.InternalError(message: "Failed to read CString, unable to read byte from \(self)")
-            }
-            guard b[0] != 0 else {
-                guard let string = String(bytes: bytes, encoding: .utf8) else {
-                    throw BSONError.InternalError(message: "Failed to decode CString as UTF8: \(bytes)")
+        let string = try self.getCString(at: self.readerIndex)
+        self.moveReaderIndex(forwardBy: string.utf8.count + 1)
+        return string
+    }
+
+    /// Attempts to read null terminated UTF-8 string from ByteBuffer starting at the offset
+    internal func getCString(at offset: Int) throws -> String {
+        let key = try getBSONKey(at: offset).dropLast()
+        guard let string = String(bytes: key, encoding: .utf8) else {
+            throw BSONError.InternalError(message: "Failed to decode BSONKey as UTF8: \(key)")
+        }
+        return string
+    }
+
+    /// Returns the C String key including the null byte, for ease of comparison and byte counting
+    internal func getBSONKey(at offset: Int) throws -> [UInt8] {
```

Comment:
The docstring mentions this is needed for "ease of comparison and byte counting", but I don't actually see where either of those are occurring. Does this need to be a separate method? It only seems to be used in `readCString`.